src/de/muenchen/allg/itd51/wollmux/event/handlers/BasicEvent.java

```diff
@@ -0,0 +1,223 @@
+package de.muenchen.allg.itd51.wollmux.event.handlers;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.star.awt.XWindow;
+import com.sun.star.frame.XFrame;
+import com.sun.star.frame.XFrames;
+import com.sun.star.uno.RuntimeException;
+
+import de.muenchen.allg.afid.UNO;
+import de.muenchen.allg.itd51.wollmux.ModalDialogs;
+import de.muenchen.allg.itd51.wollmux.WollMuxFehlerException;
+import de.muenchen.allg.itd51.wollmux.core.util.L;
+
+/**
+ * Dient als Basisklasse für konkrete Event-Implementierungen.
+ */
+public abstract class BasicEvent implements WollMuxEvent
+{
+
+  private static final Logger LOGGER = LoggerFactory
+		      .getLogger(BasicEvent.class);
+  private boolean[] lock = new boolean[] { true };
+
+  /**
+   * Diese Method ist für die Ausführung des Events zuständig. Nach der Bearbeitung
+   * entscheidet der Rückgabewert ob unmittelbar die Bearbeitung des nächsten
+   * Events gestartet werden soll oder ob das GUI blockiert werden soll bis das
+   * nächste actionPerformed-Event beim EventProcessor eintrifft.
+   */
+  @Override
+  public void process()
+  {
+    LOGGER.debug("Process WollMuxEvent " + this.toString());
+    try
+    {
+      doit();
+    }
+    catch (WollMuxFehlerException e)
+    {
+      // hier wäre ein showNoConfigInfo möglich - ist aber nicht eindeutig auf no config zurückzuführen
+      errorMessage(e);
+    }
+    // Notnagel für alle Runtime-Exceptions.
+    catch (Throwable t)
+    {
+      LOGGER.error("", t);
+    }
+  }
+
+  /**
+   * Logged die übergebene Fehlermeldung nach Logger.error() und erzeugt ein
+   * Dialogfenster mit der Fehlernachricht.
+   */
+  protected void errorMessage(Throwable t)
+  {
+    LOGGER.error("", t);
+    String msg = "";
+    if (t.getMessage() != null) msg += t.getMessage();
+    Throwable c = t.getCause();
+    /*
+     * Bei RuntimeExceptions keine Benutzersichtbare Meldung, weil
+     *
+     * 1. der Benutzer damit eh nix anfangen kann
+     *
+     * 2. dies typischerweise passiert, wenn der Benutzer das Dokument geschlossen
+     * hat, bevor der WollMux fertig war. In diesem Fall will er nicht mit einer
+     * Meldung belästigt werden.
+     */
+    if (c instanceof RuntimeException) return;
+
+    if (c != null)
+    {
+      msg += "\n\n" + c;
+    }
+    ModalDialogs.showInfoModal(L.m("WollMux-Fehler"), msg);
+  }
+
+  /**
+   * Jede abgeleitete Event-Klasse sollte die Methode doit redefinieren, in der die
+   * eigentlich event-Bearbeitung erfolgt. Die Methode doit muss alle auftretenden
+   * Exceptions selbst behandeln, Fehler die jedoch benutzersichtbar in einem
+   * Dialog angezeigt werden sollen, können über eine WollMuxFehlerException nach
+   * oben weitergereicht werden.
+   */
+  protected void doit() throws WollMuxFehlerException
```

Comment:
die Methode abstract machen, da wir uns in einer abstrakten Klasse befinden, wenn eh jede abgeleteitete Klasse die Methode redefinieren soll